### PR TITLE
chore: Allow custom default log filter

### PR DESCRIPTION
**Pull request by mkmik**
This PR moves the definition of the `log_filter` default value from the `structopt` annotation
to the `trogging::Builder`. This allows users of `trogging` to define their own default value
for the log level.

### CODE REVIEW

File: trogging/src/cli.rs [hunk 1]
```diff
@@ -20,8 +20,8 @@ pub struct TracingConfig {
     ///
     /// Extended syntax provided by `tracing-subscriber` includes span/field
     /// filters. See <https://docs.rs/tracing-subscriber/0.2.17/tracing_subscriber/filter/struct.EnvFilter.html> for more details.
-    #[structopt(long = "--log-filter", env = "LOG_FILTER", default_value = "warn")]
-    pub log_filter: String,
+    #[structopt(long = "--log-filter", env = "LOG_FILTER")]
+    pub log_filter: Option<String>,
```
Comment (crepererum) [inline trogging/src/cli.rs:24]: What happens if this option isn't passed? Does it emit ALL logs (including `trace`)? I think there is some interaction w/ the other arguments (e.g. verbosity) but I think the priority / interaction should be document here (e.g. what happens when you pass `-vvv --log-filter=warn`?)

**Review by crepererum — APPROVED** [review]
The change itself looks good, but I'm missing some docs (that are actually missing in `main`, so that's not really change-specific).

File: trogging/src/cli.rs (hunk 1 above)
Comment (mkmik) [reply trogging/src/cli.rs:24]: I _think_ `-vvv` and `--log-filter=warn` is covered in the doc for `-v` below:

```
Overrides `--log-filter`.
```

----

Yeah, the behavior in `main` specified a default value `"warn"` in the structopt annotation, which meant the default was rendered by `--help` also visible by a cursory glance in this source.

This PR buries this default in `trogging::Builder` (overrideable). Adding a doc here would render it on `--help`, which means this is not really a good place to mention code internals (which would be meaningless to the end user).

Any suggestions?

Comment (mkmik) [reply trogging/src/cli.rs:29]: Added some docs that make sense when read with rustdoc but slightly less sense when read with `--help`. Punting this for when iox CLI users will outnumber us.

**Pull request merged**